commands/commands.go

```diff
@@ -97,6 +97,75 @@ func init() {
 		help: "Gives an inspirational quote",
 		exec: inspireCommand,
 	}
+
+	commMap["minecraft"] = &command{
+		help: "Check to see who is currently online on the NetsocCraft server",
+		exec: minecraftCommand,
+	}
+}
+
+// minecraftCommand checks the stats of minecraft.netsoc.co for that moment
+// data comes from http://minecraft.netsoc.co/standalone/dynmap_NetsocCraft.json
+
+func minecraftCommand(ctx context.Context, s *discordgo.Session, m *discordgo.MessageCreate, _ []string) error {
+	l, loggerOk := logging.FromContext(ctx)
+	if loggerOk {
+		l.Infof("Responding to minecraft command", nil)
+	}
+
+	resp, err := http.Get("http://minecraft.netsoc.co/standalone/dynmap_NetsocCraft.json")
+
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve data from the Minecraft Server: %v", err)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("Failed to read the response body: %v", err)
+	}
+
+	// Unmarshal the data and get all the player lists I guess
+	q := &struct {
+		CurrentCount   int     `json:"currentcount"`
+		HasStorm       bool    `json:"hasStorm"`
+		IsThundering   bool    `json:"isThundering"`
+		ConfigHash     int     `json:"confighash"`
+		ServerTime     int     `json:"servertime"`
+		TimeStamp      int     `json:"timestamp"`
+		Players        []struct {
+			// Players nested JSON structure
+			World      string  `json:"world"`
+			Armour     int     `json:"armor"`
+			Name       string  `json:"name"`
+			X          float64 `json:"x"`
+			Y          float64 `json:"y"`
+			Z          float64 `json:"z"`
+			Sort       int     `json:"sort"`
+			Type       string  `json:"type"`
+			Account    string  `json:"account"`
+		} `json:"players"`
+		Updates        []interface{} `json:"updates"`
+	}{}
+
+	if err := json.Unmarshal(body, q); err != nil {
+		return fmt.Errorf("Failed to parse response json %q: %v", string(body), err)
+	}
+
+	// Create a discord message containing a list of all the players currently online
+	msg := "```markdown\n"
+	for i, player := range q.Players {
+		msg += fmt.Sprintf("%d. %s\n", i+1, player.Name)
+	}
+	msg += "```"
+
+	// Attempt to send the message to the discord
+	if _, err = s.ChannelMessageSend(m.ChannelID, msg); err != nil {
```

Comment:
That should be "_, err := s.Cha....". When you do an if statement like that, the declared variables are defined in the scope of the if block. The way you have it now is overwriting the already defined "err". 

It's not a problem here, but the scope leak can lead to weird bugs.

https://play.golang.org/p/yEJ27--WKn for a tasty example :P

Reply:
I just copy pasted most of this from the inspireCommand :p

Reply:
Bug in that one as well :( :( https://github.com/UCCNetworkingSociety/Netsoc-Discord-Bot/blob/master/commands/commands.go#L134

I will change this :P

Reply:
Ah okay, must have missed that one lol

Thanks for that :P